storage/aptosdb/src/lib.rs

```diff
@@ -1217,9 +1217,30 @@ impl AptosDB {
         self.ledger_db.write_set_db().write_schemas(batch)
     }
 
+    pub fn commit_genesis_ledger_info(&self, genesis_li: &LedgerInfoWithSignatures) -> Result<()> {
+        let _timer = OTHER_TIMERS_SECONDS
+            .with_label_values(&["commit_genesis_ledger_info"])
+            .start_timer();
+        let ledger_batch = SchemaBatch::new();
+        let current_epoch = self
+            .ledger_store
+            .get_latest_ledger_info_option()
+            .map_or(0, |li| li.ledger_info().next_block_epoch());
+        ensure!(genesis_li.ledger_info().epoch() == current_epoch && current_epoch == 0);
+        self.ledger_store
+            .put_ledger_info(genesis_li, &ledger_batch)?;
+
+        ledger_batch.put::<DbMetadataSchema>(
+            &DbMetadataKey::LedgerCommitProgress,
+            &DbMetadataValue::Version(0),
+        )?;
+
+        self.ledger_db.metadata_db().write_schemas(ledger_batch)
```

Comment:
How does it work with the truncator? If immediately restarted, doe it go away?

Reply:
should be